lib/Conversion/TorchToLinalg/TorchToLinalg.cpp

```diff
@@ -1828,13 +1828,13 @@ static Value createLinalgPayloadCalculationForElementwiseOp(
     Type dtype = converter->convertType(mulScalar.getType())
                      .cast<RankedTensorType>()
                      .getElementType();
-    if (!dtype.isa<mlir::FloatType>()) {
-      mulScalar.emitError("unimplemented: non-floating point dtype");
-      return nullptr;
+    Value lhs = payloadArgs[0];
+    Value rhs = convertScalarToDtype(b, loc, operands[1], dtype);
+    if (dtype.isa<mlir::FloatType>()) {
+      return b.create<arith::MulFOp>(loc, lhs, rhs);
+    } else {
```

Comment:
Please check for type, there may be types other than float and integer. https://pytorch.org/docs/stable/tensors.html

Reply:
Add a test case where types are not the same i.e., `torch.mul(x, 4.0)` where x is torch.int64.